vsphere/resource_vsphere_host.go

```diff
@@ -267,6 +314,22 @@ func resourceVsphereHostCreate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error while toggling maintenance mode for host %s. Error: %s", hostID, err)
 	}
 
+	mutableKeys := map[string]func(*schema.ResourceData, interface{}, interface{}, interface{}) error{
```

Comment:
Seems a little bit overkill for a single property.
I assume that you intend for this to be reused for other attributes in the future.
If that's the case it's okay, for a single attribute though I think it would be better to keep it simple and just handle changes to "services" explicitly

Reply:
i plan to add more services in the future for SSH, ESXi Shell, PTP. So the code will have additions, i can do that as part of this PR if that would be better